### PR TITLE
fix: PHP class-not-found WSOD + admin lockout — resilient autoloader + fail-soft bootstrap (#325)

### PR DESCRIPTION
**Pull request by parhumm**
Fixes the root cause of the #325 WSOD + admin-lockout (a class missing from the shipped classmap fataled every request, including wp-login).

- **Autoloader resilience:** `build:autoload` → `-o` (non-authoritative), `config.autoloader-suffix` pinned, `vendor/composer/autoload_real.php` regenerated — the only vendor line dropped is `setClassMapAuthoritative(true)`. A classmap miss now falls back to PSR-4 instead of fataling.
- **Runtime fail-soft:** `try/catch(\Throwable) → wp_slimstat::log` around every unconditional every-request src-class load — `RestApiManager::load_controllers` (per controller) + `register_routes` (per controller), `init()`'s `run()`, `ConsentHandler`, the whole `init_plugin()` body, and `render_gdpr_banner` (also on `login_footer`). Hardened the `gdpr_theme_mode` read.
- **Tests:** `classmap-completeness` (source-level CI gate — every `src/` class classmapped + loader non-authoritative), `gdpr-banner-theme-mode`, `Integration/RestApiFailSoftTest`, `e2e/gdpr-login-lockout` (`@compat`). CI `dump-autoload` → `-o` so the gate can't false-fail on the 8.1–8.4 lanes.

Issue #325.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved site stability so a failed component no longer takes down the whole page or blocks admin login.
  * REST/API and GDPR banner loading now fail more gracefully, reducing white-screen and lockout issues.
  * Fixed GDPR banner theme behavior when the setting is missing, preventing display issues and warnings.
* **Tests**
  * Added new regression coverage for login, REST, and GDPR banner edge cases.

### CODE REVIEW

File: wp-slimstat.php [hunk 1]
```diff
@@ -835,20 +844,27 @@ public static function slimstat_shortcode($_attributes = '', $_content = '')
...
-        add_action('template_redirect', [\SlimStat\Tracker\Tracker::class, 'adblocker_javascript']);
-        add_action('init', [\SlimStat\Tracker\Tracker::class, 'rewrite_rule_tracker']);
+            // Initialize adblock bypass functionality
+            \SlimStat\Tracker\Tracker::rewrite_rule_tracker();
+            add_action('template_redirect', [\SlimStat\Tracker\Tracker::class, 'adblocker_javascript']);
+            add_action('init', [\SlimStat\Tracker\Tracker::class, 'rewrite_rule_tracker']);
+        } catch (\Throwable $e) {
+            self::log('SlimStat: init_plugin failed to fully initialize: ' . $e->getMessage(), 'error');
+        }
     }
```
[the remark this answers is not in the document]
Comment (parhumm) [reply wp-slimstat.php:882]: Good catch — fixed in 3dea5137. Split `init_plugin()` into four isolated `try/catch` blocks (Browscap, upload dir, IP-hash salt, adblock) with step-specific log messages, so one step's failure no longer skips the others.

File: src/Providers/RestApiManager.php [hunk 2]
```diff
@@ -44,14 +44,24 @@ public static function run(): void
...
+            static function () { return new ConsentChangeRestController(); },
+            static function () { return new ConsentHealthRestController(); },
+            static function () { return new TrackerHealthRestController(); },
+        ];
+        foreach ($factories as $factory) {
+            try {
+                $controllers[] = $factory();
+            } catch (\Throwable $e) {
+                \wp_slimstat::log('SlimStat: a core REST controller failed to load: ' . $e->getMessage(), 'error');
+            }
```
[the remark this answers is not in the document]
Comment (parhumm) [reply src/Providers/RestApiManager.php:63]: Skipping this one: `RestApiManager::run()` (and therefore both catch blocks) only ever executes from within `wp_slimstat::init()`, so the global `wp_slimstat` class is guaranteed loaded at these sites — `\wp_slimstat::log()` cannot be unavailable here, making the proposed guard dead code. `log()` itself is a thin, non-throwing `error_log` wrapper. Happy to add the guard if you'd prefer belt-and-suspenders, but it wouldn't change behaviour.

**Comment by parhumm** [comment]
## 🧪 How to test — for a (non-technical) beta tester

**What this fixes:** if any part of SlimStat ever fails to load — a bad update, a half-uploaded copy, or stale server caching — the plugin now switches off just that one piece instead of taking your **whole site and your wp-admin login** down with a blank "critical error" page. This checklist just confirms everyday use still works perfectly on the pages the fix touches.

> ⚠️ Please test on a **staging/test site, not production**, and take a backup first.

**Set up**
- [ ] Update to this version and **Activate** SlimStat.

**Front-end + tracking**
- [ ] Browse 3–4 pages of your site — each page loads normally (no blank page).
- [ ] In wp-admin, open **SlimStat → Access Log** and confirm those visits were recorded.

**Admin + login page**
- [ ] These SlimStat pages all open with no error: **Real-Time**, **Access Log**, **Settings**.
- [ ] Open an **incognito / private window** and go to `your-site.com/wp-login.php` — the login form appears normally.

**GDPR banner (important — it also runs on the login page)**
- [ ] If you use SlimStat's cookie-consent banner, turn it on under **SlimStat → Settings → Tracker** (*GDPR Compliance Mode*) and save.
- [ ] In an incognito window, reload your home page → the consent banner appears; then reload `your-site.com/wp-login.php` → the login form still appears normally (no blank page).
- [ ] Set GDPR Compliance Mode back to your preferred value and save.

**Result**
- [ ] ✅ Every page above loaded and your visits were recorded → **pass**.
- [ ] ❌ Any blank / "critical error" page, or a missing login form → note **which step** and your **WordPress + PHP versions**.

---

## 🤖 How to test — prompt for an AI coding agent

```
Goal: verify the issue #325 fix — a missing/unloadable SlimStat class must
degrade gracefully, never white-screening the site or the wp-login page.

Environment: a disposable WordPress (wp-env / WordPress Playground / Local) with
THIS branch of wp-slimstat active, running on PHP 8.x, WP_DEBUG on.

1. Confirm the shipped autoloader is non-authoritative (PSR-4 fallback kept):
   `grep -c 'setClassMapAuthoritative(true)' vendor/composer/autoload_real.php`
   must be 0.

2. Run the regression suite (all must pass):
   - `composer test:source-level`  (includes test:classmap-completeness,
     test:rest-api-fail-soft, test:gdpr-banner-theme-mode)
   - `composer test:unit` and `composer test:integration`  (PHP 8.1+)

3. Baseline healthy state — request the front page and (logged out) /wp-login.php.
   Assert both return HTTP 200 with no "Fatal error" / "There has been a critical
   error" in the body.

4. Simulate the incident (the fail-soft path):
   a. Enable the consent banner: set slimstat_options gdpr_enabled=on and
      use_slimstat_banner=on so the banner renders on wp_footer AND login_footer.
   b. Temporarily make one every-request class unloadable — e.g. rename
      src/Services/Browscap.php (loaded in init_plugin) or a REST controller
      under src/Controllers/Rest/.
   c. Request the front page AND /wp-login.php (logged out). Assert BOTH still
      return HTTP 200 and render (page body / login form present) — no WSOD.
   d. Confirm wp-content/debug.log contains a "SlimStat: ... failed:" line naming
      the degraded feature (e.g. "Browscap init failed").
   e. Restore the renamed file; re-request both pages → still 200.

Report a table: surface (front page, wp-login+banner) × {baseline, missing-class,
restored} → HTTP status + fatal? (yes/no). Pass = no fatal in any cell.
```

**Comment by parhumm** [comment]
Integrated into **`dev-v551`** via merge commit `1185049f` (branch tip `da7610d4`). All checks were green on this PR (Tier-1 PHP 7.4–8.4, source-level, E2E, PHPStan). Closing since the changes now live on `dev-v551` — reopen/retarget if it needs to go to `development` too.
